packages/hawtio/src/plugins/shared/tree/node.ts

```diff
@@ -113,15 +121,14 @@ export class MBeanNode implements TreeViewDataItem {
   create(name: string, folder: boolean): MBeanNode {
     // this method should be invoked on a folder node
     if (this.children === undefined) {
-      log.warn(`node "${this.name}" should be a folder`)
```

Comment:
Just for checking my understanding, why is this warning removed?  When I added this `if` block, it was the assumption that this `create` method is not expected to be invoked on a leaf mbean node and thus `children` should be always there. That's why I added this log as warning. Has that assumption changed in any way?

Reply:
The assumption has not changed but rather the logic of node creation does not quite work. I noticed that in the `sampleapp` a node (folder) is created but it has no children close by hence it is created as a node rather than a folder. However, later in the tree reading sequence the children are then added.

Consquently, this warning represents a false positive since the node is correctly converted to a folder and the children added correctly. If you need further details, I can did out which node it is occurring on.

Reply:
Ah, I see. I think it is the case when we have two mbeans with names like `domain:aaa=xxx,bbb=yyy` and `domain:aaa=xxx,bbb=yyy,ccc=zzz`. In this case, at the first mbean `bbb=yyy` node is created as a leaf, but then at the second one it is considered as a folder since it has a child `ccc=zzz`.

The whole process of tree generation needs to be revisited with such an edge case at some point. For now, I agree with removing the log line.